store/tikv/coprocessor.go

```diff
@@ -305,9 +305,11 @@ const minLogCopTaskTime = 300 * time.Millisecond
 // The worker function that get a copTask from channel, handle it and
 // send the result back.
 func (it *copIterator) work(ctx goctx.Context, taskCh <-chan *copTask) {
+	childCtx, cancel := goctx.WithCancel(ctx)
+	defer cancel()
 	defer it.wg.Done()
```

Comment:
Put this line at the first line is better because it is much more important than `defer cancel()`.